proof_test.go

```diff
@@ -571,106 +572,162 @@ func TestVerkleProofMarshalUnmarshalJSON(t *testing.T) {
 }
 
 func TestStatelessDeserialize(t *testing.T) {
-	root := New()
-	for _, k := range [][]byte{zeroKeyTest, oneKeyTest, fourtyKeyTest, ffx32KeyTest} {
-		root.Insert(k, fourtyKeyTest, nil)
-	}
-
-	proof, _, _, _, _ := MakeVerkleMultiProof(root, keylist{zeroKeyTest, fourtyKeyTest}, map[string][]byte{string(zeroKeyTest): fourtyKeyTest, string(fourtyKeyTest): fourtyKeyTest})
-
-	serialized, statediff, err := SerializeProof(proof)
-	if err != nil {
-		t.Fatalf("could not serialize proof: %v", err)
+	insertKVs := map[string][]byte{
+		string(zeroKeyTest):   fourtyKeyTest,
+		string(oneKeyTest):    fourtyKeyTest,
+		string(fourtyKeyTest): fourtyKeyTest,
+		string(ffx32KeyTest):  fourtyKeyTest,
 	}
+	proveKeys := keylist{zeroKeyTest, fourtyKeyTest}
 
-	dproof, err := DeserializeProof(serialized, statediff)
-	if err != nil {
-		t.Fatalf("error deserializing proof: %v", err)
-	}
+	testSerializeDeserializeProof(t, insertKVs, proveKeys)
+}
 
-	droot, err := TreeFromProof(dproof, root.Commit())
-	if err != nil {
-		t.Fatal(err)
+func TestStatelessDeserializeMissingChildNode(t *testing.T) {
```

Comment:
that's nice, but there should be another one in which `oneKeyTest` is the missing key, so that we can check this use case is also working.

Reply:
Done, it's named `TestStatelessDeserializeAbsentValueInExistingLeafNode(...)`.